index.js

```diff
@@ -175,9 +175,50 @@ const plugin = fp(async function (app, opts) {
   }
 
   let entityResolversFactory
+  let gatewayRetryIntervalTimer = null
+
+  const retryServices = (interval) => {
+    let retryCount = 0
+    let shouldRetry = true
+
+    try {
+      return setInterval(async () => {
+        if (retryCount === 10) shouldRetry = false
+        retryCount++
+
+        const context = assignApplicationLifecycleHooksToContext({}, fastifyGraphQl[kHooks])
+        const serviceInfo = await gateway.refresh(shouldRetry)
+        if (!serviceInfo) return
+
+        const { schema = null, failedMandatoryServices = [] } = serviceInfo
+        if (!failedMandatoryServices.length) {
+          clearInterval(gatewayRetryIntervalTimer)
+        }
+        if (schema !== null) {
+          // Trigger onGatewayReplaceSchema hook
+          if (context.onGatewayReplaceSchema !== null) {
+            await onGatewayReplaceSchemaHandler(context, { instance: app, schema })
+          }
+          fastifyGraphQl.replaceSchema(schema)
+        }
+      }, interval)
+    } catch (error) {
+      app.log.error(error)
+    }
```

Comment:
This creates a memory leak in case of an error. Move the try catch inside the setInterval function.